packages/datadog_flutter_plugin/lib/src/tracing/tracing_headers.dart

```diff
@@ -213,24 +214,27 @@ Map<String, Object?> generateDatadogAttributes(
 
 Map<String, String> getTracingHeaders(
   TracingContext context,
-  TracingHeaderType headersType,
-) {
+  TracingHeaderType headersType, {
+  TraceContextInjection contextInjection = TraceContextInjection.all,
+}) {
   var headers = <String, String>{};
 
   final sampledString = context.sampled ? '1' : '0';
+  bool shouldInjectHeaders =
+      context.sampled || contextInjection == TraceContextInjection.all;
 
   switch (headersType) {
     case TracingHeaderType.datadog:
-      if (context.sampled) {
+      if (shouldInjectHeaders) {
         headers[DatadogHttpTracingHeaders.traceId] =
             context.traceId.asString(TracingIdRepresentation.lowDecimal);
         headers[DatadogHttpTracingHeaders.tags] =
             '${DatadogHttpTracingHeaders.traceIdTag}=${context.traceId.asString(TracingIdRepresentation.highHex16Chars)}';
         headers[DatadogHttpTracingHeaders.parentId] =
             context.spanId.asString(TracingIdRepresentation.decimal);
+        headers[DatadogHttpTracingHeaders.origin] = 'rum';
+        headers[DatadogHttpTracingHeaders.samplingPriority] = sampledString;
       }
-      headers[DatadogHttpTracingHeaders.origin] = 'rum';
-      headers[DatadogHttpTracingHeaders.samplingPriority] = sampledString;
       break;
     case TracingHeaderType.b3:
       if (context.sampled) {
```

Comment:
I think we can simplify B3 headers but that needs to be done across the board, not related to this work stream. Basically sending all the headers like we do for datadog headers because currently when a trace is not sampled and trace injection is all

the behavior is going to something like below when all trace headers are enabled.
```
        B3 = 0;
        Traceparent = "00-668e538e00000000f1b3bab076ddd791-3464ad4c72428ec2-00";
        Tracestate = "dd=p:3464ad4c72428ec2;s:0";
        "X-Amzn-Trace-Id" = "Root=1-668e538e-3ce2240c32d121f82c7fc8d3";
        "X-B3-Sampled" = 0;
        "X-Datadog-Parent-Id" = 3775332931489402562;
        "X-Datadog-Sampling-Priority" = 0;
        "X-Datadog-Tags" = "_dd.p.tid=668e538e00000000";
        "X-Datadog-Trace-Id" = 17416469451043100561;
```
I don't think anyone would go this far to have a case like this far tbh.